core/lib/scheduler.rb

```diff
@@ -23,10 +23,22 @@ def self.search(params, schedules, schedule=[])
         end
       end
     else
+      schedule.find_sort_values()
       schedules << schedule
     end
   end
 
+  def self.sort_schedules(sort_parameter, schedules)
```

Comment:
The actual sorting should be done in `ScheduleResource` (see https://www.graphiti.dev/cookbooks/without-activerecord#sorting). Additionally, avoid using "magic numbers". Rather, allow two sort parameters `average_start` and `average_finish`. If you look at the graphiti documentation, you can interpret this as follows:

step 1) capture the sorting parameters passed to the API:
```
def base_scope
  { sort: {} }
end

sort_all do |scope, attribute, direction|
  scope[:sort].merge!(attribute: attribute, direction: direction)
end
```
Here, `attribute` will ultimately be one of your sort parameters, `average_start` or `average_end`.

2) You can handle the actual sorting in the resolve method:
```
  def resolve scope
    schedules = Schedule.where scope
    if scope[:sort].present?
      schedules.sort { |s| ... }
    end
  end
```

Once these two parts are implemented, you can 'magically' call the sort function from the frontend as described here: https://www.graphiti.dev/js/reads/sorting

e.g.,

```
    Schedule
      .where({ section_id: sectionIds})
      .order({average_start: 'desc'}
```